core/mock/writer.go

```diff
@@ -12,29 +12,29 @@ import (
 	"github.com/stretchr/testify/mock"
```

Comment:
should this be split into MockWriter and MockReader?

Reply:
Since the `Writer` implements `Reader`, for mocking purposes it seems fine to combine? What do you think?

Reply:
How about having MockReader embedded into MockWriter just like how Writer embeds Reader?